src/main/java/com/hermesworld/ais/galapagos/changes/impl/ChangeBase.java

```diff
@@ -338,6 +346,79 @@ protected boolean isEqualTo(ChangeBase other) {
 
 }
 
+@JsonSerialize
+final class TopicProducerAddChange extends ChangeBase {
+
+    private final String topicName;
+
+    private final List<String> topicProducerIds;
+
+    public TopicProducerAddChange(String topicName, List<String> topicProducerIds) {
+        super(ChangeType.TOPIC_PRODUCER_APPLICATION_ADDED);
+        this.topicName = topicName;
+        this.topicProducerIds = topicProducerIds;
+
+    }
+
+    public String getTopicName() {
+        return topicName;
+    }
+
+    public List<String> getTopicProducerIds() {
+        return topicProducerIds;
+    }
+
+    @Override
+    protected boolean isEqualTo(ChangeBase other) {
+        return Objects.equals(topicName, ((TopicProducerAddChange) other).topicName);
```

Comment:
This is definitely not enough for change equality. Producer IDs must be included in comparison.